adapter/internal/discovery/xds/rate_limiter_cache.go

```
@@ -357,6 +357,24 @@ func AddSubscriptionLevelRateLimitPolicy(policyList *types.SubscriptionPolicyLis
 			rlsPolicyCache.metadataBasedPolicies[subscriptionPolicyType][policy.Organization] = make(map[string]*rls_config.RateLimitDescriptor)
 		}
 		rlsPolicyCache.metadataBasedPolicies[subscriptionPolicyType][policy.Organization][policy.Name] = descriptor
+
+		if policy.RateLimitCount > 0 && policy.RateLimitTimeUnit != "" {
+			burstCtrlUnit, err := parseRateLimitUnitFromSubscriptionPolicy(policy.RateLimitTimeUnit)
+			if err != nil {
+				loggers.LoggerXds.Error("Error while getting the burst control time unit", err)
```

Let's return the error.